src/OpenTelemetry.Exporter.Geneva/JsonSerializer.cs

```diff
@@ -0,0 +1,289 @@
+        {
+            if (array == null)
+            {
+                return SerializeNull(buffer, cursor);
+            }
+
+            buffer[cursor++] = unchecked((byte)'[');
+            var length = array.Length;
+            if (length >= 1)
+            {
+                cursor = Serialize(buffer, cursor, array[0]);
+                for (int i = 1; i < length; i++)
+                {
+                    buffer[cursor++] = unchecked((byte)',');
+                    cursor = Serialize(buffer, cursor, array[i]);
+                }
+            }
+
+            buffer[cursor++] = unchecked((byte)']');
+            return cursor;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static string SerializeMap(IEnumerable<KeyValuePair<string, object>> map)
+        {
+            var count = SerializeMap(buffer.Value, 0, map);
+            return Encoding.UTF8.GetString(buffer.Value, 0, count);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int SerializeMap(byte[] buffer, int cursor, IEnumerable<KeyValuePair<string, object>> map)
+        {
+            if (map == null)
+            {
+                return SerializeNull(buffer, cursor);
+            }
+
+            buffer[cursor++] = unchecked((byte)'{');
+            int count = 0;
+            foreach (var entry in map)
+            {
+                if (count > 0)
+                {
+                    buffer[cursor++] = unchecked((byte)',');
+                }
+
+                cursor = SerializeString(buffer, cursor, entry.Key);
+                buffer[cursor++] = unchecked((byte)':');
+                cursor = Serialize(buffer, cursor, entry.Value);
+                count++;
+            }
+
+            buffer[cursor++] = unchecked((byte)'}');
+            return cursor;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static string Serialize(object obj)
+        {
+            var count = Serialize(buffer.Value, 0, obj);
+            return Encoding.UTF8.GetString(buffer.Value, 0, count);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int Serialize(byte[] buffer, int cursor, object obj)
+        {
+            if (obj == null)
+            {
+                return SerializeNull(buffer, cursor);
+            }
+
+            switch (obj)
+            {
+                case bool v:
+                    return WriteString(buffer, cursor, v ? "true" : "false");
+                case byte:
+                case sbyte:
+                case short:
+                case ushort:
+                case int:
+                case uint:
+                case long:
+                case ulong:
+                case float:
+                case double:
+                case DateTime:
+                    return WriteString(buffer, cursor, Convert.ToString(obj, CultureInfo.InvariantCulture));
```

Comment:
Add a TODO comment here, we should be able to use something better - e.g. https://github.com/open-telemetry/opentelemetry-dotnet/blob/a37198c6d0f1814f8f5a995413649436657a2e2b/src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs#L46-L54